clang-tools-extra/clang-tidy/performance/UnnecessaryCopyInitialization.cpp

```diff
@@ -261,21 +262,27 @@ void UnnecessaryCopyInitialization::registerMatchers(MatchFinder *Finder) {
                      this);
 }
 
+UnnecessaryCopyInitialization::CheckContext::CheckContext(
```

Comment:
I'm not a fan of this constructor, it should exist.
instead object of CheckContext should be created as:

```
CheckContext context{*NewVar, *BlockStmt, *OldVar, ...};
```

simply moving functionality of check into helper class isn't nice.
```

Reply:
Given that we are deriving `IssueFix`, `IsVarUnused`, `IsVarOnlyUsedAsConst` from `Var`, `BlockStmt`, ... it sounds natural to have a ctor there. Otherwise we risk creating a `CheckContext` in an invalid state.

Reply:
Problem is that code were moved from check method to CheckContext constructor. And thats not nice. CheckContext (context by name) shouldn't have any logic.